app/views/layouts/angular-bootstrap/_auth_credentials_angular_bootstrap.html.haml

```diff
@@ -17,8 +17,30 @@
 - vm_scope               ||= false
 - validate_button        ||= true
 - main_scope            = vm_scope ? "$parent.vm" : "$parent"
+- ng_show_service_account ||= false
+- service_account_label   ||= _("Service account")
 
 %div{"ng-controller" => "CredentialsController as vm", "vm-scope" => "$parent.vm.model ? $parent.vm : $parent"}
+
+  %div{"ng-show" => "#{ng_show} && #{ng_show_service_account}"}
+    .form-group{"ng-class" => "{'has-error': angularForm.service_account.$invalid}"}
+      %label.col-md-2.control-label{"for" => "#{prefix}_service_account"}
+        = service_account_label
+      .col-md-4
+        - hash_for_service_account = {"type"                    => "text",
+                                      "id"                      => "#{prefix}_service_account",
+                                      "ng-required"             => false,
+                                      "ng-disabled"             => false,
+                                      "name"                    => "service_account",
```

Comment:
`id` and `name` must match,  and they should also match the field name in `ng-model`, otherwise, you're breaking assumptions in the code.

(And then what @AparnaKarve said about `params[:default_service_account]` is true.)

Reply:
done